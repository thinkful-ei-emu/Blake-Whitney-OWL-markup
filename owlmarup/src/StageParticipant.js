import React from 'react';

function StageParticipant(props) {
// props.person.name, props.person.avatar

  return (
    <div className="on-stage-participant">
      <div className="on-stage-participant-name">
        <span>{props.person.name}</span>
        <div className="mic-bar"></div>
        <img src={props.person.avatar} alt="person avatar"></img>
      </div>
    </div>

  );

}

export default StageParticipant;