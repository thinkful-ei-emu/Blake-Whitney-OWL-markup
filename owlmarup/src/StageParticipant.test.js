import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import StageParticipant from './Stageparticipant';

describe('<StageParticipant />', () => {
  // Smoke Test
  it('renders without crashing', () => {
    // Create a DOM element to render the component into
    const div = document.createElement('div');

    // Render the component
    // If something is wrong it will fail here
    ReactDOM.render(<StageParticipant person={{
      id: 1,
      name: 'Koren Templeton',
      avatar:
          'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1',
      inSession: true,
      onStage: true
  }}/>, div);

    // Clean up
    ReactDOM.unmountComponentAtNode(div);
  });

  // Snapshot Test
  it('renders this UI as expected', () => {
    const tree = renderer.create(<StageParticipant person={{
      id: 1,
      name: 'Koren Templeton',
      avatar:
          'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1',
      inSession: true,
      onStage: true
  }}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});