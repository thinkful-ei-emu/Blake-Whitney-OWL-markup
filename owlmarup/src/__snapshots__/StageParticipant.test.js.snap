// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StageParticipant /> renders this UI as expected 1`] = `
<div
  className="on-stage-participant"
>
  <div
    className="on-stage-participant-name"
  >
    <span>
      Koren Templeton
    </span>
    <div
      className="mic-bar"
    />
    <img
      alt="person avatar"
      src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
    />
  </div>
</div>
`;
